cornell-dti/office-hours
# Collect student feedback

### Summary 


- provide a means for students to give feedback on the TA who just finished helping them
- prompt students with a feedback modal that pops up after the TA marks their question as done
- enable both numeric star feedback and qualitative feedback the student writes
- create Firebase functions to save this feedback object into the course document
- feedback object stores OH session, question ID, rating, and written feedback

Demo of what the entire process looks like, with the prompt appearing properly and the student's input appearing in Firestore

https://github.com/cornell-dti/office-hours/assets/100158738/47afc68d-0a77-4e56-9961-09826e7849cf

Below, we have a rudimentary version of what the feedback modal will look like


<img width="968" alt="Screen Shot 2023-11-15 at 12 59 45 AM" src="https://github.com/cornell-dti/office-hours/assets/100158738/839e6eff-791a-4f68-acd6-087037797ea8">







### Documentation
included in this PR as specification comments

### Test Plan 
- open two terminals, one as a TA and the other as a student, and see that the feedback modal properly appears when question is marked as done
- check that the feedback data properly appears on the Firebase console with the associated course



### Notes 

The styling in this PR is definitely a work in progress. Substitutions for the star rating in MUI's icon library that look more similar to the design or just larger in general weren't found, but will keep looking for other ways.

Will need to discuss form validation – e.g. whether or not we're able to submit incomplete feedback and the associated warnings that may pop up if we don't have sufficient feedback given.


Also note that this is a New PR for [this ticket](https://www.notion.so/Star-rating-after-each-question-space-for-feedback-50f53284f184457badf24b5a72202de6) without all the annoying linting issues from [this PR](https://github.com/cornell-dti/office-hours/pull/761) 

### Breaking Changes 

None
